src/extractor/guidance/turn_handler.cpp

```diff
@@ -476,108 +587,153 @@ std::pair<std::size_t, std::size_t> TurnHandler::findFork(const EdgeID via_edge,
             best = i;
         }
     }
-    if (best_deviation <= NARROW_TURN_ANGLE)
-    {
-        std::size_t left = best, right = best;
-        while (
-            left + 1 < intersection.size() &&
-            (angularDeviation(intersection[left + 1].angle, STRAIGHT_ANGLE) <= NARROW_TURN_ANGLE ||
-             (angularDeviation(intersection[left].angle, intersection[left + 1].angle) <=
-                  NARROW_TURN_ANGLE &&
-              angularDeviation(intersection[left].angle, STRAIGHT_ANGLE) <= GROUP_ANGLE)))
-            ++left;
-        while (
-            right > 1 &&
-            (angularDeviation(intersection[right - 1].angle, STRAIGHT_ANGLE) <= NARROW_TURN_ANGLE ||
-             (angularDeviation(intersection[right].angle, intersection[right - 1].angle) <
-                  NARROW_TURN_ANGLE &&
-              angularDeviation(intersection[right - 1].angle, STRAIGHT_ANGLE) <= GROUP_ANGLE)))
-            --right;
-
-        if (left == right)
-            return std::make_pair(std::size_t{0}, std::size_t{0});
-
-        const bool valid_indices = 0 < right && right < left;
-        const bool separated_at_left_side =
-            angularDeviation(intersection[left].angle,
-                             intersection[(left + 1) % intersection.size()].angle) >= GROUP_ANGLE;
-        const bool separated_at_right_side =
-            right > 0 &&
-            angularDeviation(intersection[right].angle, intersection[right - 1].angle) >=
-                GROUP_ANGLE;
-
-        const bool not_more_than_three = (left - right) <= 2;
-        const bool has_obvious = [&]() {
-            if (left - right == 1)
-            {
-                return isObviousOfTwo(via_edge, intersection[left], intersection[right]) ||
-                       isObviousOfTwo(via_edge, intersection[right], intersection[left]);
-            }
-            else if (left - right == 2)
+    return std::make_tuple(best, best_deviation);
+}
+
+std::pair<std::size_t, std::size_t>
+TurnHandler::findLeftAndRightmostForkCandidates(const Intersection &intersection) const
+{
+    std::size_t best;
+    double best_deviation;
+    std::tie(best, best_deviation) = findClosestToStraight(intersection);
+
+    // no fork found
+    if (best_deviation > NARROW_TURN_ANGLE)
+    {
+        return std::make_pair(std::size_t{0}, std::size_t{0});
```

Comment:
Same here - semantically this is wrong, isn't it? The index 0 is the UTurn road. Shouldn’t this return an `optional<T>` and `T` being a `struct { leftmost, rightmost }`?